### ISSUE TITLE
Episode 3: Creating a challenge using mkdir -p Instructor checkout

### ISSUE THREAD

**Issue by gregomics** [swcarpentry/shell-novice#890]
Please delete the text below before submitting your contribution. 

---

We could introduce mkdir -p to create an architecture with parent with child directory. It would refresh the nested architecture and show the "No such file or directory” error. 
We can create a challenge and show it ether by a doing a double mkdir commands or one command mkdir -p. Here could be the challenge:

Nelle is in her home directory and wants to create a directory `courses/` where she can store each different topic into another sub directory as the following example:
`/Users/nelle/courses/math`

How can she proceed?

1. `mkdir courses` 
	`cd courses`
	`mkdir math`

2. `mkdir courses/math`

3. `mkdir courses`
	`mkdir courses/math`

4. `mkdir -p courses/math`

### Solutions 

1. Yes the first set of commands will work and produce the requested architecture

2. No, this will produce an error : “mkdir: cannot create directory ‘courses/math’: No such file or directory”. The parent directory courses/ is not yet created.

3. Yes, this will produce the requested directories. 

4. Yes, this will produce directly the parent and child directories.

**Comment by colinmorris** [comment]
I'm a -1 on teaching the `-p` flag. It's a nice convenience to be aware of if you write a lot of shell scripts, but I don't think it offers enough value for shell beginners to be worth spending time on.

I do like the idea of giving them some basic practice in creating directories. I think your proposed exercise would work well to that end if option 4 was replaced with a distractor like...

```
mkdir courses
mkdir math
```
That said, episode 3 already has 10 exercises, so I'm not sure if this meets the bar for adding yet another.

**Comment by gcapes** [comment]
That's a fair enough rationale I guess.

What do you think about the exercise `copy a folder structure but not the files`? http://swcarpentry.github.io/shell-novice/03-create/index.html#copy-a-folder-structure-but-not-the-files

I think it's a clunky horrible solution to the problem, and not a great workflow to be recommending -- imagine if there are lots of files to be copied only to then be deleted. How about rewording the question to challenge the learner to recreate the folder structure using `mkdir` commands?

**Comment by gdevenyi** [comment]
I'm +1 on the ``-p`` flag, but I think it could be used to salvage #897 

Thanks @gregomics, if we decide to introduce the ``-p`` option, we'll address it on #897
